aws/sdk/examples/sts/src/bin/credentials-provider.rs

```diff
@@ -14,10 +15,16 @@ async fn main() -> Result<(), dynamodb::Error> {
     tracing_subscriber::fmt::init();
     let client = sts::Client::from_env();
 
-    // NOTE: Do not use LazyCachingCredentialsProvider in production yet!
-    // It hasn't implemented timeout or panic safety yet.
+    // `LazyCachingCredentialsProvider` will load credentials if it doesn't have any non-expired
+    // credentials cached. See the docs on the builder for the various configuration options,
+    // such as timeouts, default expiration times, and more.
     let sts_provider = LazyCachingCredentialsProvider::builder()
-        .refresh(async_provide_credentials_fn(move || {
+        // `LazyCachingCredentialsProvider` requires an implementation of async sleep so that
+        // it can be used with different async runtimes. The sleep function will be used for timeouts.
+        // Since we're using Tokio in this example, we use the `TokioSleep` implementation, which
+        // requires the `timeout-tokio` feature in the `aws-auth` dependency.
+        .sleeper(TokioSleep::new())
```

Comment:
I agree w/ @rcoh that, in the default case, this should not be specified by the user.